contracts/cw-croncat/src/tasks.rs

```diff
@@ -201,6 +202,31 @@ impl<'a> CwCroncat<'a> {
         })
     }
 
+    /// Query returns gas estimation and number of occurrences by TaskRequest
+    pub(crate) fn query_simulate_task(
+        &self,
+        env: Env,
+        deps: Deps,
+        task: TaskRequest,
+        funds: GenericBalance,
+    ) -> StdResult<SimulateTaskResponse> {
+        let cfg: Config = self.config.load(deps.storage)?;
+
+        let sim = simulate_task(
+            task,
+            funds,
+            cfg.gas_base_fee,
+            cfg.gas_action_fee,
+            env.block,
+            cfg.slot_granularity_time,
+        )
+        .unwrap();
```

Comment:
Let's have useful error if something failed during validation!
```suggestion
        .map_err(|e| cosmwasm_std::StdError::generic_err(e.to_string()))?;
```